### DISCUSSION TITLE
A take on FreeBSD14.1

### DISCUSSION THREAD

**Discussion by paaguti** [orbitalquark/textadept#562] [General]
Hi

I'm trying to get TextAdept to run on FreeBSD (again).

My target is a FreeBSD 14.1 with KDE5/plasma5. In my initial take I'm including
`-D QT=ON -D GTK2=OFF -D GTK3=OFF -D CURSES=OFF`

With that, I'm able to compile, but then I get a core...
I'd like to start sharing experiences here ;-)

**Comment by orbitalquark** [comment]
On startup, Textadept needs to figure out where its home is. On BSD, it used to do this: https://github.com/orbitalquark/textadept/commit/bb14dfe31c87ebf9b1489186c7a023ef6582d952#diff-e137203d90aa39a746e4bae862aff9bcd13a78d4f4fe2896b0046aba666f9ed5

You can try restoring that code first and see where that takes you.

**Reply by paaguti** [reply]
It took me to a functional Textadept 12.5 nightly on FreeBSD14.1 with KDE plasma5:

![image](https://github.com/user-attachments/assets/f3c269f6-4db8-45fc-90fd-7d4b2e8feb44)

I'll continue testing, but in the meantime, would it be possible to restore the BSD detection code into the repositories, please?

Thanks in advance :-)

**Reply by paaguti** [reply]
As a matter of fact, the Qt version has a lot less dependencies, so it's quite simple to build.

**Comment by M0JXD** [comment]
Just adding some additional notes to build for FreeBSD 15, all in one place for the next person who wants to build it. At the moment Qt6 builds don't start due to something with SingleApplication, but Qt5 works.

### 1. GTK won't link

Depending on your BSD you might have to specify link directories for GTK, you can add this early in the CMakeLists file like below. See #607 

```cmake
# FreeBSD link directory
link_directories(/usr/local/lib)
```

### 2. Building with GCC

If you specify GCC over Clang with the CMake flag -DCMAKE_C_COMPILER, the linker might not be able to find iconv. You can fix this like so for the platforms you wish to build:

```cmake
# You can do this early in the CMakeLists.txt to discover iconv
find_package(Iconv REQUIRED)

# Add in the library  Textadept platform 
if(PLATFORM_NAME)
	add_library(textadept_plat OBJECT src/textadept_plat.c)
	target_include_directories(textadept_plat PRIVATE ${PLAT_INCLUDE_DIRS})
	# Ditto ...

	# Force iconv discovery on FreeBSD for GCC
	target_link_libraries(textadept_plat PRIVATE Iconv::Iconv)
endif()
```
